content/4_Gold/Intro_Bitwise.mdx

```diff
@@ -71,50 +71,146 @@ Now that we can acquire the sum of any two numbers in two queries, we can easily
+	int and_ = ask("and", ++a, ++b);
+	int or_ = ask("or", a, b);
+
+	int xor_ = ~and_ & or_;  // a ^ b = ~(a & b) & (a | b)
+
+	return 2 * and_ + xor_;  // a + b = 2(a & b) + a ^ b
 }
 
 int main() {
 	int n, k;
 	cin >> n >> k;
 
-	// acquire first 3 elements
-	int aplusb = sum(1, 2);
-	int bplusc = sum(2, 3);
-	int aplusc = sum(1, 3);
-
-	// solve a system of equations
-	vector<int> a(3);
-	a[1] = (aplusb + bplusc - aplusc) / 2;
-	a[0] = aplusb - a[1];
-	a[2] = bplusc - a[1];
-
-	// get the rest of the array
-	for (int i = 4; i <= n; i++) { a.push_back(sum(i - 1, i) - a.back()); }
-	sort(a.begin(), a.end());
-	cout << "finish " << a[k - 1] << endl;
+	// Acquire the first 3 elements
+	int a_plus_b = sum(0, 1);
+	int a_plus_c = sum(0, 2);
+	int b_plus_c = sum(1, 2);
+
+	// Get the actual values by solving the equations
+	vector<int> arr{(a_plus_b + a_plus_c - b_plus_c) / 2};
+	arr.push_back(a_plus_b - arr[0]);
+	arr.push_back(a_plus_c - arr[0]);
+
+	// Get the rest of the array
+	for (int i = 3; i < n; i++) { arr.push_back(sum(i - 1, i) - arr.back()); }
+
+	sort(arr.begin(), arr.end());
+	cout << "finish " << arr[k - 1] << endl;
 }
 ```
+
 </CPPSection>
-</LanguageSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class TakeAGuess {
+	private static BufferedReader br =
+	    new BufferedReader(new InputStreamReader(System.in));
+
+	public static void main(String[] args) throws IOException {
+		StringTokenizer st = new StringTokenizer(br.readLine());
+		int n = Integer.parseInt(st.nextToken());
+		int k = Integer.parseInt(st.nextToken());
+
+		// Acquire the first 3 elements
+		int aPlusB = sum(0, 1);
+		int aPlusC = sum(0, 2);
+		int bPlusC = sum(1, 2);
+
+		// Get the actual values by solving the equations
+		List<Integer> arr = new ArrayList<>();
+		arr.add((aPlusB + aPlusC - bPlusC) / 2);
+		arr.add(aPlusB - arr.get(0));
+		arr.add(aPlusC - arr.get(0));
+
+		// Get the rest of the array
+		for (int i = 3; i < n; i++) {
+			arr.add(sum(i - 1, i) - arr.get(arr.size() - 1));
+		}
 
-Now let's take a look on how to implement addition and multiplication using only bitwise operators.
+		arr.sort(Comparator.comparingInt(i -> i));
+		System.out.printf("finish %d%n", arr.get(k - 1));
+	}
+
+	/** @return the sum of the elements at a and b (0-indexed) */
+	private static int sum(int a, int b) throws IOException {
+		int and = ask("and", ++a, ++b);
+		int or = ask("or", a, b);
+
+		int xor = ~and & or;  // a ^ b = ~(a & b) & (a | b)
+
+		return 2 * and + xor;  // a + b = 2(a & b) + a ^ b
+	}
+
+	private static int ask(String s, int a, int b) throws IOException {
+		System.out.printf("%s %d %d%n", s, a, b);
+		System.out.flush();
+		return Integer.parseInt(br.readLine());
+	}
+}
+```
+
+</JavaSection>
+<PySection>
+
+```py
+def ask(s: str, a: int, b: int) -> int:
+	print(f"{s} {a} {b}", flush=True)
+	return int(input())
+
+
+def sum(a: int, b: int) -> int:
+	""":return the sum of the elements at a and b (0-indexed)"""
```

Comment:
wait shouldn't u put this above function decl?

Reply:
no, doc comments are in the function